specs/tls.rst

```diff
@@ -0,0 +1,201 @@
+  it does not understand.  Critical fields that are not understood cause
+  certificate parsing to fail).
+* allows the client to do wierd and wonderful things with the raw certificate,
+  instead of placing arbitrary limits on it.
+
+Specifying ``tls_version`` as an integer, not a string or float:
+
+* avoids maintenance effort in this specification. If a new version of TLS is
+  defined, then no changes are needed in this specification.
+* does not significantly affect servers.  Whatever format we specified, servers
+  would likely need a lookup table from what their TLS library reports to what
+  this API needs.  (Unless their TLS library provides access to the raw value,
+  in which case it can be reported via this API directly).
+* does not significantly affect clients.  Whatever format we specified, clients
+  would likely need a lookup table from what this API reports to the values
+  they support and wish to use internally.
+
+Specifying ``cipher_suite`` as an integer, not a string:
+
+* avoids significant effort to compile a list of cipher suites in this
+  specification.  There are a huge number of existing TLS cipher suites, many
+  of which are not widely used, even listing them all would be a huge effort.
+* avoids maintenance effort in this specification. If a new cipher suite is
+  defined, then no changes are needed in this specification.
+* avoids dependencies on nonstandard TLS-library-specific names.  E.g. the
+  cipher names used by OpenSSL are different from the cipher names used by the
+  RFCs.
+* does not significantly affect servers.  Whatever format we specified, (unless
+  it was a nonstandard library-specific name and the server happened to use
+  that library), servers would likely need a lookup table from what their
+  TLS library reports to what this API needs.  (Unless their TLS library
+  provides access to the raw value, in  which case it can be reported via this
+  API directly).
+* does not significantly affect clients.  Whatever format we specified, clients
+  would likely need a lookup table from what this API reports to the values
+  they support and wish to use internally.
+* Using a single integer, rather than a pair of integers, makes handling this
+  value simpler and faster.
+
+``client_cert_name`` duplicates information that is also available in
+``client_cert_chain``.  However, many ASGI applications will probably find
+that information is sufficient for their application - it provides a simple
+string that identifies the user.  It is simpler to use than parsing the x509
+certificate.  For the server, this information is readily available.
+
+There are theoretical interoperability problems with ``client_cert_name``,
+since it depends on a list of object ID names that is maintained by IANA and
+theoretically can change.  In practise, this is not a real problem, since the
+object IDs that are actually used in certificates have not changed in many
+years.  So in practise it will be fine.
```

Comment:
practise -> practice

Reply:
Thanks, should be fixed now.  I ran spellcheck this time, too!